src/mg/PAMI/Message/IncomingMessage.php

```diff
@@ -81,6 +87,33 @@ public function getRawContent()
     }
 
     /**
+     * Returns the channel variables for all reported channels.
+     * https://github.com/marcelog/PAMI/issues/85
+     *
+     * The channel names will be lowercased.
+     *
+     * @return array
+     */
+    public function getAllChannelVariables()
+    {
+      return $this->channelVariables;
+    }
+
+    /**
+     * Returns the channel variables for the given channel.
+     * https://github.com/marcelog/PAMI/issues/85
+     *
+     * @param string $channel Channel name. If not given, will return variables
+     * for the "current" channel.
+     *
+     * @return array
+     */
+    public function getChannelVariables($channel = 'default')
```

Comment:
It would be cool also if @johanwilfer could try this branch and validate it in a real world scenario :)